pkg/retry/azure_error.go

```diff
@@ -327,47 +327,85 @@ func HasStatusForbiddenOrIgnoredError(err error) bool {
 	return false
 }
 
-// ParseRawError parse the error message in the rawError and unmarshal it into RawErrorContainer
-func ParseRawError(rawError string) (*RawErrorContainer, error) {
-	reg := regexp.MustCompile(`^(?:[^{]*)([\s\S]*)$`)
-	matches := reg.FindStringSubmatch(rawError)
-	if len(matches) != 2 {
-		klog.V(4).Infof("skipping parsing because the format of the raw error message %q is not the expected one")
-		return nil, nil
+// GetVMSSMetadataByRawError gets the vmss name by parsing the error message
+func GetVMSSMetadataByRawError(err *Error) (string, string, error) {
+	if err == nil || !isErrorLoadBalancerInUseByVirtualMachineScaleSet(err.RawError.Error()) {
+		return "", "", nil
 	}
 
-	rawErrorMap := make(map[string]*RawErrorContainer)
-	err := json.Unmarshal([]byte(matches[1]), &rawErrorMap)
-	if err != nil {
-		return nil, err
+	reg := regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.*)/providers/Microsoft.Compute/virtualMachineScaleSets/(.+).`)
+	matches := reg.FindStringSubmatch(err.ServiceErrorMessage())
```

Comment:
That's the change for the load balancer error parsing logic. We don't need the `RawErrorContainer` since its doing the unnecessary round trip of converting to a string and building into a local struct.